src/Anonymization/Anonymizer/Core/IntegerAnonymizer.php

```diff
@@ -23,18 +26,67 @@ class IntegerAnonymizer extends AbstractAnonymizer
      */
     public function anonymize(Update $update): void
     {
-        if (!($this->options->has('min') && $this->options->has('max'))) {
-            throw new \InvalidArgumentException("You should provide 2 options (min and max) with this anonymizer");
+
+        if ($this->options->has('min') && $this->options->has('max')) {
+            $this->anonymizeWithMinAndMax(
+                $update,
+                (int) $this->options->get('min'),
+                (int) $this->options->get('max')
+            );
+        } elseif ($this->options->has('delta')) {
+            $this->anonymizeWithDelta($update, (int) $this->options->get('delta'));
+        } elseif ($this->options->has('percent')) {
+            $this->anonymizeWithPercent($update, (int) $this->options->get('percent'));
+        } else {
+            throw new \InvalidArgumentException("You should provide options with this anonymizer: min and max, or delta, or percent");
         }
+    }
 
+    private function anonymizeWithMinAndMax(Update $update, int $max, int $min): void
+    {
         $update->set(
             $this->columnName,
             $this->getSetIfNotNullExpression(
-                $this->getRandomIntExpression(
-                    $this->options->get('max'),
-                    $this->options->get('min'),
-                ),
+                $this->getRandomIntExpression($max, $min),
             ),
         );
     }
+
+    private function anonymizeWithDelta(Update $update, int $delta): void
+    {
+        $expr = $update->expression();
+
+        $update->set(
+            $this->columnName,
+            $expr->raw(
+                '? + ?',
+                [
+                    $expr->column($this->columnName, $this->tableName),
+                    $this->getRandomIntExpression($delta, -$delta)
+                ]
+            )
+        );
+    }
+
+    private function anonymizeWithPercent(Update $update, int $percent): void
+    {
+        $expr = $update->expression();
+
+        $update->set(
+            $this->columnName,
+            $expr->cast(
+                $expr->raw(
+                    '? * (?) / 100',
```

Comment:
I wonder about the parentheses. Are they really useful?

Reply:
yep, it does not work without them

Reply:
Quite strange :thinking:

Reply:
Well i guess it depends of what the `$this->getRandomIntExpression` returns (in SQL). 

It only failed for Mysql I think.

Reply:
No I think it was SQLite :

```php

        return $this->formatRaw(
            new Raw(
                'abs(random()) % (? - ?) + ?',
                [$max, $min, $min]
            ),
            $context,
        );
```
(from the query builder source code)

Reply:
:+1:

Reply:
Finally, shouldn't we fix this method from the `SqliteWriter` instead?

Reply:
see https://github.com/makinacorpus/php-query-builder/issues/6

Reply:
> I agree but as it comes from a dependency, I suggest we keep it like this for now

Very true :sweat_smile: